### PR TITLE
[CDC] Support cdc sink to UNAWARE bucket mode table

### PR DESCRIPTION
**Pull request by TaoZex**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

**Comment by TaoZex** [comment]
The failed tests are all due to timeout errors in the kafka container. Please retry the ci.

**Comment by yuzelin** [comment]
I think 'Support cdc sink to UNAWARE bucket mode table' is more precise.

**Comment by TaoZex** [comment]
Thanks for the suggestion, I have changed the title

> I think 'Support cdc sink to UNAWARE bucket mode table' is more precise.

Thanks for the suggestion, I have changed the title.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/CdcUnawareBucketWriteSink.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.apache.paimon.flink.sink.StoreSinkWrite;
+import org.apache.paimon.flink.sink.UnawareBucketSink;
+import org.apache.paimon.table.AppendOnlyFileStoreTable;
+
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+
+import java.util.Map;
+
+/** CDC Sink for unaware bucket table. */
+public class CdcUnawareBucketWriteSink extends UnawareBucketSink {
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/CdcUnawareBucketWriteSink.java:31]: extends UnawareBucketSink<CdcRecord>

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/CdcUnawareBucketWriteSink.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    public CdcUnawareBucketWriteSink(AppendOnlyFileStoreTable table, Integer parallelism) {
+        super(table, null, null, parallelism, true);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/sink/cdc/CdcUnawareBucketWriteSink.java:34]: I think boundedInput should be false.

File: paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/sink/cdc/TestTable.java [hunk 3]
```diff
@@ -125,14 +130,12 @@ public TestTable(
...
 
                 List<CdcRecord> records = new ArrayList<>();
-                boolean shouldInsert = true;
                 if (expected.containsKey(key)) {
                     records.add(new CdcRecord(RowKind.DELETE, expected.get(key)));
-                    expected.remove(key);
-                    // 20% chance to only delete without insert
-                    shouldInsert = random.nextInt(5) > 0;
-                }
-                if (shouldInsert) {
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/sink/cdc/TestTable.java:135]: Here we can retain the old test method. If `isAppendTable  == false`, use the old logic; else use the new logic

Comment (TaoZex) [reply paimon-flink/paimon-flink-cdc/src/test/java/org/apache/paimon/flink/sink/cdc/TestTable.java]: done.

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**